include/Graph.hpp

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include <iostream>
+#include <string>
+#include <vector>
+
+class Node {
+ public:
+  int abc;
+  std::vector<Node*> prevNodes;
+  std::vector<Node*> nextNodes;
+  std::string Type;
+  Node(std::string type) : Type(type) {}
+  void addPrev(Node* node) { prevNodes.push_back(node); }
+  void addNext(Node* node) { nextNodes.push_back(node); }
+  bool isConnected(Node* otherNode) {
+    for (Node* node : nextNodes) {
+      if (node == otherNode) {
+        return true;
+      }
+    }
+    for (Node* node : prevNodes) {
+      if (node == otherNode) {
+        return true;
+      }
+    }
+    return false;
+  }
+  bool isNext(Node* otherNode) {
+    for (Node* node : nextNodes) {
+      if (node == otherNode) {
+        return true;
+      }
+    }
+    return false;
+  }
+  bool isPrev(Node* otherNode) {
+    for (Node* node : prevNodes) {
+      if (node == otherNode) {
+        return true;
+      }
+    }
+    return false;
+  }
+};
+
+class Graph {
```

Comment:
Please use adjacency list as simple form of graph

Reply:
Sorry. I don't quite understand what you mean. Do you want me to use zeros and ones and output them for clarity? Or is it important to have some zero at the end of the list, which I decided not to use here? Or is it necessary to organize a line-by-line output that includes the names of Nodes with subsequent redefinition of cout? Or should I go back a few commits and assign Nodes to IDs to get arrays of IDs?

Reply:
Or is it just necessary to change vector to list?

Reply:
https://en.wikipedia.org/wiki/Adjacency_list this representation of graph maybe

Reply:
I advice to use this article https://arxiv.org/pdf/1903.10036.pdf (page 3 has scheme of structure)

Reply:
also we can use this structure for understanding 
![Uploading Screenshot 2023-12-06 at 20.25.53.png…]()